src/test/java/net/darkhax/teslatest/item/ItemBattery.java

```diff
@@ -0,0 +1,144 @@
+package net.darkhax.teslatest.item;
+
+import net.darkhax.tesla.api.BaseTeslaContainer;
+import net.darkhax.tesla.capability.TeslaCapabilities;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.nbt.NBTBase;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.util.EnumFacing;
+import net.minecraftforge.common.capabilities.Capability;
+import net.minecraftforge.common.capabilities.ICapabilityProvider;
+import net.minecraftforge.common.util.INBTSerializable;
+
+import java.util.List;
+/**
+ * A Item that can accept power and also have it pulled out. The
+ * goal of this Item is to have a tooltip of the energy stored, and store energy. This
+ * class is also used as the default example of how to write a Item that can use Tesla
+ * power.
+ */
+public class ItemBattery extends Item {
+
+    // An instance of something that implements ITeslaConsumer, ITeslaProducer or
+    // ITeslaHandler. In this case we use the BaseTeslaContainer which makes use of all three.
+    // The purpose of this instance is to handle all tesla related logic for the Item.
+
+    public ItemBattery() {
+        this.setCreativeTab(CreativeTabs.TOOLS);
+        this.setUnlocalizedName("battery");
+        this.setMaxStackSize(1);
+    }
+
+    @Override
+    public int getDamage(ItemStack stack) {
+        if(stack.hasCapability(TeslaCapabilities.CAPABILITY_HOLDER,EnumFacing.DOWN))
+            return (int) stack.getCapability(TeslaCapabilities.CAPABILITY_HOLDER,EnumFacing.DOWN).getStoredPower();
+        else
+            return super.getDamage(stack);
+    }
+
+    @Override
+    public int getMaxDamage(ItemStack stack) {
+        if(stack.hasCapability(TeslaCapabilities.CAPABILITY_HOLDER,EnumFacing.DOWN))
+            return (int) stack.getCapability(TeslaCapabilities.CAPABILITY_HOLDER,EnumFacing.DOWN).getCapacity()+1;
+        else
+            return super.getMaxDamage(stack);
+    }
+
+    /*
+     * Sets item tooltip saying how much energy is stored and capacity
+     */
+    @Override
+    public void addInformation(ItemStack stack, EntityPlayer playerIn, List<String> tooltip, boolean advanced) {
+        super.addInformation(stack, playerIn, tooltip, advanced);
+        tooltip.add(I18n.format("label.teslatest.battery.normal",stack.getCapability(TeslaCapabilities.CAPABILITY_HOLDER,EnumFacing.DOWN).getStoredPower(),stack.getCapability(TeslaCapabilities.CAPABILITY_HOLDER,EnumFacing.DOWN).getCapacity()));

```

Comment:
Might be best to store the CAPABILITY_HOLDER would save this long line, and speed things up a tiny bit